### ISSUE TITLE
Routing hooks error

### ISSUE THREAD

**Issue by Wayfindertm** [krasimir/navigo#281]
```JS
router.hooks({
	before(done, match) {
		console.log('Before hook');
		router.navigate("/auth", {callHooks: false});
		done();
	}
});
router
	.on('/', () => {
		console.log('Home');
	})
	.on('/auth', () => {
		console.log('Auth');
	});
```

And when we call home route we see in console:
```
Before hook
Home
Auth
```

How can a force open auth without calling home?

**Comment by krasimir** [comment]
You should not call the `done` callback in the `before` hook. This basically means "keep going and resolve the routes". You either don't call it or you call it with `false` (`done(false)`). Here's a test case that shows how it works https://github.com/krasimir/navigo/blob/master/src/__tests__/hooks.spec.ts#L58-L74

**Comment by Wayfindertm** [comment]
So if i remove done(), router.navigate will not work?

**Comment by krasimir** [comment]
Hm ... interesting. Can you try with `done(false)`.

**Comment by Wayfindertm** [comment]
Before hook

Thats all (((

**Comment by krasimir** [comment]
Sorry, I did not look at your code in details. So what is happening is that you are defining a hook for every route. Including `/auth`. Which means that you successfully navigate to it but the before hook kicks in because we are doing `done(false)` we can't continue and call the handler. The proper implementation will look like so:

```javascript
router
  .on('/auth', () => {
    console.log('Auth');
  });
  .on('/', () => {
    console.log('Home');
  }, {
    before(done) {
      router.navigate("/auth", {callHooks: false});
      done(false);
    }
  })
```

**Comment by Wayfindertm** [comment]
Your code dosn't work too

And this code too:
`router.hooks({
	before(done, match) {
		console.log(match.url);
		if (match.url !== 'auth') {
			console.log('Before hook');
			router.navigate("/auth");
			done(false);
		}
	}
});`

**Comment by Wayfindertm** [comment]
I need to make something like middleware to check auth, so i want use global hook for all routes except auth

**Comment by krasimir** [comment]
I see. let me come with a test case.

**Comment by krasimir** [comment]
Hey @Wayfindertm,

there was a bug which is fixed in a new `8.10.0` version. Can you please try it and let me know how it works. Here's the test case:

```typescript
describe("and we want to navigate out from a hook", () => {
  it("should properly land on the new path", () => {
    const r: NavigoRouter = new Navigo();
    const h1 = jest.fn();
    const h2 = jest.fn();
    r.hooks({
      before(done, match) {
        if (match.url !== "auth/sign-in") {
          r.navigate("/auth/sign-in");
          done(false);
        } else {
          done();
        }
      },
    });
    r.on("/auth/sign-in", h1);
    r.on("/", h2);

    r.resolve();

    expect(h1).toBeCalledTimes(1);
    expect(h2).not.toBeCalled();
  });
});

```

**Comment by Wayfindertm** [comment]
It works great!!! Thanx!!!
